### PR TITLE
Remove necessity of declaring PIN_APP_EXTENSIONS macro

### PR DESCRIPTION
**Pull request by maicki**

### CODE REVIEW

File: PINCache/PINDiskCache.m [hunk 1]
```diff
@@ -1107,10 +1107,16 @@ - (void)unlock
 @end
 
 @implementation PINBackgroundTask
+
++ (BOOL)isAppExtension
+{
+    return [[[NSBundle mainBundle] executablePath] containsString:@".appex/"];
```
Comment (garrettmoon) [inline PINCache/PINDiskCache.m]: should this be dispatch_once'd?

Comment (benasher44) [inline PINCache/PINDiskCache.m]: Can you write a unit test for this? I think this is doable if you separated this into 2 methods, once of which just took a string and return a BOOL, and `isAppExtension` could  all that one with `[[NSBundle mainBundle] executablePath]`

**Comment by benasher44** [comment]
Here's the smoke test I thought of to help ensure this can't break again:

- Add a new extension target to PINCache.xcodeproj
- Add a new test target that tests the extension target
- Add a test that just initializes the cache objects and tries to call the `start` method, which in a non-extension context would create a background task. If this breaks, the test should crash/fail.

**Comment by benasher44** [comment]
Does this sound realistic/doable?

File: PINCache/PINDiskCache.m (hunk 1 above)
Comment (maicki) [inline PINCache/PINDiskCache.m]: Yep this will be dispatched_once. I will look into how we could integrate a unit test for it.

**Comment by benasher44** [comment]
Also, @martin-gearzero's final comment in #68 seems preferable to inspecting any of the main bundle's paths.

**Comment by maicki** [comment]
@benasher44 @garrettmoon Added the dispatch_once for the is app extension check and added a basic share extension target that just initializes PINCache and calls a method that would create a PINBackgroundTask.

I couldn't really figure out how we can test PINCache usage within the share extension via a test target. Happy for any help!

File: PINCache/PINDiskCache.m [hunk 2]
```diff
@@ -1120,11 +1134,17 @@ - (instancetype)init
 + (instancetype)start
 {
     PINBackgroundTask *task = [[self alloc] init];
```
Comment (maicki) [inline PINCache/PINDiskCache.m]: @garrettmoon You think we should just return nil here and not allocate any PINBackgroundTask if isAppExtension == YES?

Comment (garrettmoon) [inline PINCache/PINDiskCache.m]: Seems like a good call to me.

**Comment by garrettmoon** [comment]
@maicki It might be possible to do with a UI automation test? Force the extension to run in Safari via a share extension?

I think this is a good start though and worth merging.

**Pull request merged**